src/commands/bug_report.rs

```diff
@@ -0,0 +1,408 @@
+    } else {
+        log::trace!("Failed to get container_file");
+        String::new()
+    };
+
+    BlueBuildInfo {
+        recipe,
+        container_file,
+    }
+}
+
+fn auto_complete(p: String) -> Completions<String> {
+    use std::path::Path;
+
+    let current: &Path = p.as_ref();
+    let (mut dir, last) = if p.ends_with('/') {
+        (current, "")
+    } else {
+        let dir = current.parent().unwrap_or_else(|| "/".as_ref());
+        let last = current
+            .file_name()
+            .and_then(std::ffi::OsStr::to_str)
+            .unwrap_or("");
+        (dir, last)
+    };
+
+    if dir.to_str().unwrap().is_empty() {
+        dir = ".".as_ref();
+    }
+
+    let mut files: Completions<_> = match dir.read_dir() {
+        Ok(files) => files
+            .flatten()
+            .filter_map(|file| {
+                let path = file.path();
+                let is_dir = path.is_dir();
+                match path.into_os_string().into_string() {
+                    Ok(s) if is_dir => Some(s + "/"),
+                    Ok(s) => Some(s),
+                    Err(_) => None,
+                }
+            })
+            .collect(),
+        Err(_) => {
+            return completions![p];
+        }
+    };
+
+    if files.is_empty() {
+        return completions![p];
+    }
+
+    let fuzzer = SkimMatcherV2::default();
+    files.sort_by_cached_key(|file| fuzzer.fuzzy_match(file, last).unwrap_or(i64::MAX));
+    files
+}
+
+// ============================================================================= //
+
+struct Environment {
+    shell_info: ShellInfo,
+    os_type: os_info::Type,
+    terminal_info: TerminalInfo,
+    os_version: os_info::Version,
+}
+
+#[derive(Debug)]
+struct ShellInfo {
+    name: String,
+    version: String,
+}
+
+fn get_shell_info() -> ShellInfo {
+    let failure_shell_info = ShellInfo {
+        name: UNKNOWN_SHELL.to_string(),
+        version: UNKNOWN_VERSION.to_string(),
+    };
+
+    let current_shell = match Shell::from_env() {
+        Some(shell) => shell.to_string(),
+        None => return failure_shell_info,
+    };
+
+    let version = get_shell_version(&current_shell);
+
+    ShellInfo {
+        version,
+        name: current_shell.to_string(),
+    }
+}
+
+// ============================================================================= //
+// Git
+// ============================================================================= //
+
+#[derive(Debug, Clone, Template, TypedBuilder)]
+#[template(path = "github_issue")]
+struct GithubIssueTemplate<'a> {
+    bb_version: &'a str,
+    build_rust_channel: &'a str,
+    build_time: &'a str,
+    containerfile: &'a str,
+    git_commit_hash: &'a str,
+    os_name: &'a str,
+    os_version: &'a str,
+    pkg_branch_tag: &'a str,
+    recipe: &'a str,
+    rust_channel: &'a str,
+    rust_version: &'a str,
+    shell_name: &'a str,
+    shell_version: &'a str,
+    terminal_name: &'a str,
+    terminal_version: &'a str,
```

Comment:
So this is something I've been meaning to do for the other templates except let's use `Cow` for this so it will end up being like:

```suggestion
    #[builder(setter(into))]
    bb_version: Cow<'a, str>,
	
    #[builder(setter(into))]
    build_rust_channel: Cow<'a, str>,
	
    #[builder(setter(into))]
    build_time: Cow<'a, str>,
	
    #[builder(setter(into))]
    containerfile: Cow<'a, str>,
	
    #[builder(setter(into))]
    git_commit_hash: Cow<'a, str>,
	
    #[builder(setter(into))]
    os_name: Cow<'a, str>,
	
    #[builder(setter(into))]
    os_version: Cow<'a, str>,
	
    #[builder(setter(into))]
    pkg_branch_tag: Cow<'a, str>,
	
    #[builder(setter(into))]
    recipe: Cow<'a, str>,
	
    #[builder(setter(into))]
    rust_channel: Cow<'a, str>,
	
    #[builder(setter(into))]
    rust_version: Cow<'a, str>,
	
    #[builder(setter(into))]
    shell_name: Cow<'a, str>,
	
    #[builder(setter(into))]
    shell_version: Cow<'a, str>,
	
    #[builder(setter(into))]
    terminal_name: Cow<'a, str>,
	
    #[builder(setter(into))]
    terminal_version: Cow<'a, str>,
```

This will allow passing in either owned or borrowed data into the builder